docs/book/src/reference/metrics.md

```diff
@@ -40,38 +40,163 @@ Further information can be found bellow in this document.
+because they do not establish a chain of trust, making them vulnerable to security threats.
+
+Furthermore, check the configuration file located at `config/prometheus/monitor.yaml` to
+ensure secure integration with Prometheus. If the `insecureSkipVerify: true` option is enabled,
+it means that certificate verification is turned off. This is **not** recommended for production as
+it poses a significant security risk by making the system vulnerable to man-in-the-middle attacks,
+where an attacker could intercept and manipulate the communication between Prometheus and the monitored services.
+This could lead to unauthorized access to metrics data, compromising the integrity and confidentiality of the information.
+
+**In both cases, the primary risk is potentially allowing unauthorized access to sensitive metrics data.**
+
+### Recommended Actions for a Secure Production Setup
+
+1. **Replace Self-Signed Certificates:**
+   - Instead of using `TLSOpts`, configure the `CertDir`, `CertName`, and `KeyName` options to use your own certificates.
+   This ensures that your server communicates using trusted and secure certificates.
+
+2. **Configure Prometheus Monitoring Securely:**
+   - Check and update your Prometheus configuration file (`config/prometheus/monitor.yaml`) to ensure secure settings.
+   - Replace `insecureSkipVerify: true` with the following secure options:
+
+     ```yaml
+     caFile: The path to the CA certificate file, e.g., /etc/metrics-certs/ca.crt.
+     certFile: The path to the client certificate file, e.g., /etc/metrics-certs/tls.crt.
+     keyFile: The path to the client key file, e.g., /etc/metrics-certs/tls.key.
+     ```
+
+   These settings ensure encrypted and authenticated communication between Prometheus and the monitored services, providing a secure monitoring setup.
+</aside>
+
+<aside class="note">
+<h1>Controller-Runtime Auth/Authz Feature Current Known Limitations and Considerations</h1>
+
+Some known limitations and considerations have been identified. The settings for `cache TTL`, `anonymous access`, and
+`timeouts` are currently hardcoded, which may lead to performance and security concerns due to the inability to
+fine-tune these parameters. Additionally, the current implementation lacks support for configurations like
+`alwaysAllow` for critical paths (e.g., `/healthz`) and `alwaysAllowGroups` (e.g., `system:masters`), potentially
+causing operational challenges. Furthermore, the system heavily relies on stable connectivity to the `kube-apiserver`,
+making it vulnerable to metrics outages during network instability. This can result in the loss of crucial metrics data,
+particularly during critical periods when monitoring and diagnosing issues in real-time is essential.
+
+An [issue](https://github.com/kubernetes-sigs/controller-runtime/issues/2781) has been opened to
+enhance the controller-runtime and address these considerations.
+</aside>
```

Comment:
See that via the notes we are raising awareness and let them be aware of all.
See wdyt, please feel free to tweek if you see fit.